hsmd/hsmd_wire.csv

```diff
@@ -323,3 +323,11 @@ msgdata,hsmd_derive_secret,info,u8,len
 # Reply with the derived secret
 msgtype,hsmd_derive_secret_reply,127
 msgdata,hsmd_derive_secret_reply,secret,secret,
+
+# Sanity check this pubkey derivation is correct.
+msgtype,hsmd_check_pubkey,28
+msgdata,hsmd_check_pubkey,index,u32,
```

Comment:
- are hardened indices allowed (high bit = 1)?
- we plan to deny very high indices, e.g. > 1_000_000, to make recovery sane in case of database loss.  is that OK?

Reply:
No, we don't use hardened derivations.  And the 1M keys limit seems like a sane precaution: if someone exceeds that they should tell us :)